### ISSUE TITLE
Unsupported input file format: upgrade_loader.pkg(Hisense TV)

### ISSUE THREAD

**Issue by Luupw3d** [openlgtv/epk2extract#15]
I have built  epk2extract in Ubuntu 18.04 ,but when I tried to run it , it noticed "Unsupported input file format: upgrade_loader.pkg".
I also tried to use binwalk -e to extrat it ,but faild.

MT5327

**Comment by smx-smx** [comment]
Hi! Thanks for your interest in the project 🙂 
Could you post a link to a downloadable firmare file?

If it's a Mediatek chip (you mentioned MT5327) it should be supported, unless it has a different format or it's completely encrypted.

**Comment by smx-smx** [comment]
Ok here is the problem.
Mediatek firmware updates are crypted, with the exception of the partition names.
There is no signature to detect if it's a Mediatek firmware or not, so the only way to check if it's a valid Mediatek Firmware is trying to decrypt it.

If decryption fails, there's an heuristic (that was only tested on Sharp and Philips firmwares). The heuristic tries to see if the first partition is called "cfig" or "ixml". If it matches, it will dump out the **CRYPTED** partitions and bail out (so you only get some blobs of encrypted data).

In other words, you don't have the proper encryption key to decrypt the firmware (if you had it, you wouldn't be getting that message).
Maybe i could add a message like "Trying to decrypt as Mediatek Firmware", to inform users that we tried that, and failed.

If you want to add your own key, edit "keys/MTK.key", rebuild (or edit build_<platform>/MTK.key directly), and try again.
If you happen to have root access on your TV, you can dump the key it's using to decrypt the software update. We have done that for most keys in the repository.

**Comment by smx-smx** [comment]
Nevermind i was tired/wrong. It's not encrypted and i need to fix the handling 🙂

**Comment by Luupw3d** [comment]
In fact,there is an file named Mustang_m5v1_cn_4k2k_secure_emmcboot.bin but I haven't upload it yet.Do you need it？

**Comment by smx-smx** [comment]
It shouldn't be needed, i nailed it down what is required to fix it.
I started doing some changes to support both variants but i haven't finished yet, time has been limited lately.
Either way i believe your format is older than the one epk2extract currently supports. It misses a field in the package structure.
Hopefully i'll complete it soon.

**Comment by smx-smx** [comment]
I made some preliminary change to support the format.
Partitions are split fine, and some of them can be extracted further.

I'm not sure what format sysm.pak is. It could be a filesystem image, or it could need some extra handling
